src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -3429,14 +3366,12 @@ bool AdaptDispatch::SetCursorStyle(const DispatchTypes::CursorStyle cursorStyle)
 
     default:
         // Invalid argument should be handled by the connected terminal.
-        return false;
+        break;
```

Comment:
This should `return` rather than `break`, otherwise it'll apply the legacy style which is not what we want. Comment could also do with an update.